linker/shared/src/test/scala/org/scalajs/linker/MaxModuleSplittingTest.scala

```diff
@@ -66,4 +73,48 @@ class MaxModuleSplittingTest {
       assertEquals(expectedFiles, outputDirectory.fileNames().toSet)
     }
   }
+
+  @Test // #4542
+  def noMemExplostionOnImportChain(): AsyncResult = await {
+    /* Create a long import chain. Previous versions of the linker would attempt
+     * to construct the superset of import hops.
```

Comment:
Isn't it a power set?
```suggestion
     * to construct the power set of import hops.
```